### PR TITLE
[JENKINS-73053] Allow users with Overall/Manage permission to configure endpoints

### PR DESCRIPTION
**Pull request by amuniz**
# Description

Allow users with Overall/Manage permission to configure endpoints. See [JENKINS-73053](https://issues.jenkins.io/browse/JENKINS-73053) for further information. 

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

### CODE REVIEW

**Review by olamy — APPROVED** [review]

**Review by mikecirioli — APPROVED** [review]

**Comment by jglick** [comment]
Do I get it right that this only fixes permission checks on form validation and dropdowns and that the actual submission of a new endpoint was already allowed?

**Comment by amuniz** [comment]
I was missing a `getRequiredGlobalConfigPagePermission` override. Fixed it and added a test for it.
Thanks for the catch!

**Review by joseblas — APPROVED** [review]

**Review by mikecirioli — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/github_branch_source/EndpointTest.java [hunk 1]
```diff
@@ -90,6 +91,13 @@ public void canPostAsAdmin_doCheckApiUri() throws Exception {
         assertTrue(TestRoot.get().visited);
     }
 
+    @Test
+    @Issue("JENKINS-73053")
+    public void manageCanSetupEndpoints() throws Exception {
+        HtmlPage htmlPage = j.createWebClient().login("alice").goTo("manage/configure");
+        assertTrue(htmlPage.getVisibleText().contains("GitHub Enterprise Servers"));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/github_branch_source/EndpointTest.java:98]: Does not verify that you can actually save new endpoints, but that is much harder to test in WebClient (given YUI craziness), and I guess ought to be implied by the same `getRequiredGlobalConfigPagePermission`. Did you at least try it interactively? (Most easily by just setting a breakpoint in this test and then logging in as `alice` / `alice` from a real browser.)

**Review by jglick — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/github_branch_source/EndpointTest.java (hunk 1 above)
Comment (amuniz) [reply src/test/java/org/jenkinsci/plugins/github_branch_source/EndpointTest.java:98]: Verified manually. It works as expected.

**Pull request merged**